src/pages/Login/Login.js

```diff
@@ -0,0 +1,101 @@
+import React, { useReducer, useState } from 'react'
+import './Login.css'
+import { useAuth } from 'context'
+import { loginFormReducer } from 'reducer'
+import axios from 'axios'
+import { useNavigate, Link } from 'react-router-dom'
+const Login = () => {
+    const navigate = useNavigate()
+    const { authDispatch } = useAuth();
+
+    const [{ email, password }, loginDispatch] = useReducer(loginFormReducer, { email: "", password: "" })
+
+
+    const [togglePassword, setTogglePassword] = useState(false);
+    const showPassword = togglePassword ? "visibility_off" : "visibility";
+    const passwordVisibilityHandler = () => {
+        setTogglePassword(!togglePassword)
```

Comment:
If you state update is dependent on previous value then use the functional way of updating state